### PR TITLE
PLUGIN-81 - Splunk Batch Sink

### PR DESCRIPTION
**Pull request by rostyslavmorozov**
WIKI: https://wiki.cask.co/display/CE/Splunk+Plugin
JIRA: https://issues.cask.co/browse/PLUGIN-81

Included:
- Batch Sink

### CODE REVIEW

File: src/main/java/io/cdap/plugin/splunk/common/config/BaseSplunkConfig.java [hunk 1]
```diff
@@ -152,6 +181,14 @@ public Integer getMaxRetryJitterWait() {
 
   public void validate(FailureCollector collector) {
     IdUtils.validateReferenceName(referenceName, collector);
+    try {
+      if (!containsMacro(PROPERTY_URL)) {
+        getUrl();
+      }
+    } catch (IllegalArgumentException e) {
+      collector.addFailure(e.getMessage(), null)
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/splunk/common/config/BaseSplunkConfig.java]: validation error message should contain field name

File: src/test/java/io/cdap/plugin/splunk/source/batch/SplunkBatchSourceConfigBuilder.java [hunk 2]
```diff
@@ -67,6 +69,7 @@
   private String indexedLatestTime;
   private Long searchResultsCount;
   private String schema;
+  private boolean mockConnection = false;
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/io/cdap/plugin/splunk/source/batch/SplunkBatchSourceConfigBuilder.java]: I think this hack is too dirty. Maybe it's better to move connection related validation to another method? Maybe even not in config class.

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: src/main/java/io/cdap/plugin/splunk/common/config/BaseSplunkConfig.java (hunk 1 above)
Comment (rostyslavmorozov) [reply src/main/java/io/cdap/plugin/splunk/common/config/BaseSplunkConfig.java]: Done.

File: src/test/java/io/cdap/plugin/splunk/source/batch/SplunkBatchSourceConfigBuilder.java (hunk 2 above)
Comment (rostyslavmorozov) [reply src/test/java/io/cdap/plugin/splunk/source/batch/SplunkBatchSourceConfigBuilder.java]: Done.

**Comment by rostyslavmorozov** [comment]
approved by @mlozbin-cybervisiontech

**Pull request closed without merging**